sympy/core/power.py

```diff
@@ -404,10 +404,20 @@ def _eval_Mod(self, q):
 
             '''
             For unevaluated Integer power, use built-in pow modular
-            exponentiation.
+            exponentiation, if powers are not too large wrt base.
             '''
             if self.base.is_Integer and self.exp.is_Integer and q.is_Integer:
-                return pow(int(self.base), int(self.exp), int(q))
+                b, e, m = int(self.base), int(self.exp), int(q)
+                # For very large powers, use totient reduction.
+                # Bound on m, is for safe factorization memory wise ie m^(1/4).
+                # For pollard-rho to be faster than built-in pow and for e >= lg(m),
+                # lg(e) > m^(1/4) check is added.
+                if m.bit_length() <= 80  and _log(e,2) > m**(1/4):
```

Comment:
Could `e.bit_length()` be used here?

Reply:
Or maybe `e.bit_length()**4 > m`.

Reply:
Yes, in fact, using this clubbed with `e >= m.bit_length()`, would not only be more explicit, but also more readable.

So, the final block would look like,
```
mb = m.bit_length()
if mb <= 80 and e >= mb and e.bit_length()**4 > m:
```

Reply:
This also avoids potential issues with float overflow, in case q is very large.

Reply:
@asmeurer Yes, true. So, should I commit to that?